iam_apikey.go

```diff
@@ -0,0 +1,64 @@
+package egoscale
+
+// APIKeyType holds the type of the apikey
+type APIKeyType string
+
+const (
+	// APIKeyUnrestricted is unrestricted
+	APIKeyUnrestricted APIKeyType = "restricted"
+	// APIKeyRestricted is restricted
+	APIKeyRestricted APIKeyType = "unrestricted"
+)
+
+// APIKey represents an apikey
+type APIKey struct {
+	Description string     `json:"description"`
+	Key         string     `json:"key"`
+	Secret      string     `json:"secret,omitempty"`
+	Operations  []string   `json:"operations,omitempty"`
+	Type        APIKeyType `json:"type"`
+}
+
+// CreateAPIKey represents the apikey creation
+type CreateAPIKey struct {
+	Description string `json:"description"`
+	Operations  string `json:"operations,omitempty"`
```

Comment:
I think can be nice to put a slice for this field

Reply:
I agree.

Reply:
not possible for the API at the moment.
(likely to change in the near future)